ksql-streams/src/main/java/io/confluent/ksql/execution/streams/JoinParams.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.streams;
+
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.Objects;
+
+public class JoinParams {
```

Comment:
I don't love this name. We have a few classes (`JoinParams/JoinParamsFactory`, `AggregateParams/AggregateParamsFactory`, `Selection`) that own the transformation at various steps (to the schema and the data). It makes sense that these (the transformation to the data, and to the schema) should be coupled together, but I'm unsure what to call it.